DIPs/DIP1012.md

```diff
@@ -23,28 +23,43 @@ propose any mechanism to disable compiler attributes directly (e.g. `@!nogc`).
 
 ## Rationale
 
-Many users feel that the default attributes have the wrong default behavior and, given that attributes are not invertable, putting 
+Many users feel that the default attributes have the wrong default behavior and, given that attributes are not invertable (the negations do not have names), putting 
 ```
 pure: nothrow: @nogc: @safe:
 ```
-at the top of a file means that one can never "undo" those attributes. This DIP proposes a solution to easily change the default values of the attributes while also allowing symbol-specific overrides of individual attribute groups.
+at the top of a file means that one can never "undo" those attributes (except `@safe`/`@system`/`@trusted`).
+This DIP proposes a solution to easily change the default values of the attributes by making them regular attributes enabling selected enforcement at varying levels of granularity:
+* whole program (`core.attribute.defaultAttributeSet`),
+* module scope / remainder of scope (`@attribute:`), 
+* declaration block (`@attribute { ... }`) and 
+* symbol (`@attribute void foo() { ... }`) 
+
+Regular UDAs are manipulated via `AliasSeq`s, but the compiler recognised attributes are not regular attributes and so they can't be.
+This leads to a difference between manipulating the different types of attributes and the need for explicit `__traits` to deal with the compiler attibutes.
+
+Additionally making the compiler recognised attributes regular attributes has a number of advantages:
+* manipulation of lists of attributes becomes simpler as traditional methods of `AliasSeq` manipulation can be used
+* decouples (from a metaprogramming prespective) the definition of the attributes from their usage eg. to allow listing required core attributes for a method in meta-type definition.
+* new ones can be "phased in" by using `@future` without fear of breaking code. Also having the infrastructure in place will make development of new attributes faster.
+* it is hoped that this will lower the aversion to adding new attributes
 
 ## Description
 
-Move all (DMD) compiler-recognized attributes into `core.attribute`, making them symbols in their own right. They shall be organized by attribute group into `enum`s, each with the following members:
-* `inferred`. Indicates the compiler shall determine the value of the attribute. It is illegal for a function declaration without a body to be marked `inferred`. Does not apply to `final`/`virtual`.
+Move all (DMD) compiler-recognized function attributes into `core.attribute`, making them symbols in their own right. They shall be organized by attribute category into `enum`s, each with the following members:
+* `inferred`. Indicates the compiler shall determine the value of the attribute. Function declarations marked `inferred` resolve to the default value. Does not apply where the compiler is not responsible for inference e.g. `final`/`virtual`.
 * the attribute's default value e.g. `@system` or impure
```

Comment:
The explicit default is for whole program default so that every symbol has a valid value. It won't be needed if they are not included as part of the DIP.